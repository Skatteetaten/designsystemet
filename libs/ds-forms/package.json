{
  "name": "@skatteetaten/ds-forms",
  "groupId": "no.skatteetaten.aurora",
  "version": "2.0.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^24",
    "react": "^19 ",
    "react-i18next": "^15",
    "@skatteetaten/ds-core-utils": "2.0.0",
    "@skatteetaten/ds-icons": "2.0.0",
    "@skatteetaten/ds-buttons": "2.0.0",
    "@skatteetaten/ds-typography": "2.0.0",
    "@skatteetaten/ds-progress": "2.0.0",
    "@skatteetaten/ds-status": "2.0.0",
    "date-fns": "^4",
    "@skatteetaten/ds-content": "2.0.0"
  },
  "scripts": {}
}
