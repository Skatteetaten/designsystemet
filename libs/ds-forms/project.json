{
  "name": "ds-forms",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-forms/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-forms",
        "tsConfig": "libs/ds-forms/tsconfig.lib.json",
        "project": "libs/ds-forms/package.json",
        "entryFile": "libs/ds-forms/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-forms/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-forms/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-forms"],
      "options": {
        "jestConfig": "libs/ds-forms/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-forms/**/*.css",
          "libs/ds-forms/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-forms"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-forms {args.switches}"
      }
    }
  }
}
