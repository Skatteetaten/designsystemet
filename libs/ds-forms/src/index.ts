export * from './Checkbox/Checkbox';
export * from './Checkbox/Checkbox.types';
export * from './CheckboxGroup/CheckboxGroup';
export * from './DatePicker/DatePicker.types';
export * from './DatePicker/DatePicker';
export * from './Select/Select';
export * from './Select/Select.types';
export * from './SelectOption/SelectOption.types';
export * from './CheckboxGroup/CheckboxGroup.types';
export * from './ErrorMessage/ErrorMessage';
export * from './ErrorMessage/ErrorMessage.types';
export * from './ErrorSummary/ErrorSummary';
export * from './ErrorSummary/ErrorSummary.types';
export * from './Fieldset/Fieldset';
export * from './Fieldset/Fieldset.types';
export * from './FileUploader/FileUploader';
export * from './FileUploader/FileUploader.types';
export * from './Radio/Radio';
export * from './Radio/Radio.types';
export * from './RadioGroup/RadioGroup';
export * from './RadioGroup/RadioGroup.types';
export * from './TextArea/TextArea.types';
export * from './TextArea/TextArea';
export * from './TextField/TextField.types';
export * from './TextField/TextField';
export * from './SearchField/SearchField';
export * from './SearchField/SearchField.types';
