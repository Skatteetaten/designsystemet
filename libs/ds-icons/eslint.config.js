import nxPlugin from '@nx/eslint-plugin';

import baseConfig from '../../eslint.config.js';

export default [
  ...baseConfig,
  ...nxPlugin.configs['flat/react'],
  {
    ignores: ['!**/*'],
  },
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    files: ['**/*.ts', '**/*.tsx'],
    rules: {},
  },
  {
    files: ['**/*.js', '**/*.jsx'],
    rules: {},
  },
];
