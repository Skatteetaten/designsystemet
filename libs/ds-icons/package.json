{
  "name": "@skatteetaten/ds-icons",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.6.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.6.0"
  },
  "scripts": {},
  "dependencies": {},
  "devDependencies": {}
}
