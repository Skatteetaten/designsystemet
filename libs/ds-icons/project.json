{
  "name": "ds-icons",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-icons/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-icons",
        "tsConfig": "libs/ds-icons/tsconfig.lib.json",
        "project": "libs/ds-icons/package.json",
        "entryFile": "libs/ds-icons/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-icons/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-icons/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-icons"],
      "options": {
        "jestConfig": "libs/ds-icons/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-icons/**/*.css",
          "libs/ds-icons/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-icons"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-icons {args.switches}"
      }
    }
  }
}
