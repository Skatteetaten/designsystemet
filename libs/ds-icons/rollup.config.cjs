const path = require('path');

const createRollupConfig = require('../ds-dev-config/build/rollup-config-factory.cjs');

const srcDir = path.resolve(__dirname, `src`);
const pathToTsConfig = 'libs/ds-icons/tsconfig.lib.json';
const pathToCSS = path.resolve(__dirname, `**/*.{css,scss}`);
const outputDir = path.resolve('./', `dist/libs/ds-icons/`);

module.exports = (config) =>
  createRollupConfig(config, srcDir, pathToTsConfig, pathToCSS, outputDir);
