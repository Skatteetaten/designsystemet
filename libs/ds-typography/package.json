{
  "name": "@skatteetaten/ds-typography",
  "version": "1.1.0",
  "groupId": "no.skatteetaten.aurora",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.1.0"
  },
  "scripts": {}
}
