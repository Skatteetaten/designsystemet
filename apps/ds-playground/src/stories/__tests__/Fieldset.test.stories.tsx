import { Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { Fieldset, FieldsetProps } from '@skatteetaten/ds-forms';
import { WarningSVGpath } from '@skatteetaten/ds-icons';
import { Heading, Paragraph } from '@skatteetaten/ds-typography';

import {
  loremIpsumWithoutSpaces,
  wrapper,
} from './testUtils/storybook.testing.utils';
import { SystemSVGPaths } from '../utils/icon.systems';

const meta = {
  component: Fieldset,
  title: 'Tester/Fieldset',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    classNames: { table: { disable: true } },
    children: { table: { disable: true } },
    description: { table: { disable: true } },
    hasSpacing: { table: { disable: true } },
    helpText: { table: { disable: true } },
    helpSvgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    hideLegend: { table: { disable: true } },
    legend: { table: { disable: true } },
    showRequiredMark: { table: { disable: true } },
    titleHelpSvg: { table: { disable: true } },
    // HTML
    disabled: { table: { disable: true } },
    form: { table: { disable: true } },
    // Events
    onHelpToggle: { table: { disable: true } },
  },
} satisfies Meta<typeof Fieldset>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultDescription = 'Kort hjelpetekst';
const defaultHelpText =
  'Vi trenger Ã¥ vite navnet ditt dersom vi skal kontakte deg senere.';
const defaultLegendText = 'Ledetekst';
const defaultArgs: FieldsetProps = {
  legend: defaultLegendText,
  children: <div>{'Innhold'}</div>,
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLFieldSetElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const fieldset = canvas.getByRole('group');
    await expect(fieldset).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
    form: '123form',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
    form: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const fieldset = canvas.getByRole('group');
    await expect(fieldset).toHaveAttribute('id', 'htmlid');
    await expect(fieldset).toHaveClass('dummyClassname');
    await expect(fieldset).toHaveAttribute('lang', 'nb');
    await expect(fieldset).toHaveAttribute('data-testid', '123ID');
    await expect(fieldset).toHaveAttribute('form', '123form');
  },
} satisfies Story;

export const Defaults = {
  name: 'Default (FS-A1, FS-A2, FS-A4, A2, A3, B1)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    legend: { table: { disable: false } },
    children: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const fieldset = canvas.getByRole('group');
    await expect(fieldset.tagName).toBe('FIELDSET');

    await expect(fieldset.firstElementChild?.tagName).toBe('LEGEND');
  },
} satisfies Story;

export const WithDescription = {
  name: 'With Description (FS-A3)',
  args: {
    ...defaultArgs,
    description: defaultDescription,
  },
  argTypes: {
    description: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const descriptionNode = canvas.getByText(defaultDescription);
    await expect(descriptionNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithShowRequiredMark = {
  name: 'With ShowRequiredMark (FS-A4)',
  args: {
    ...defaultArgs,
    showRequiredMark: true,
  },
  argTypes: {
    showRequiredMark: { table: { disable: false } },
  },
} satisfies Story;

export const WithShowRequiredMarkAndLegend = {
  name: 'With ShowRequiredMark And Legend Contains Markup (FS-A4)',
  args: {
    ...defaultArgs,
    legend: (
      <>
        <Heading as={'h1'} level={3}>
          {'Dette er en Heading i legend'}
        </Heading>
        <Paragraph variant={'ingress'}>
          <em>{'Dette er en italic Paragraph med ingress variant i legend'}</em>
        </Paragraph>
      </>
    ),
    showRequiredMark: true,
  },
  argTypes: {
    legend: {
      table: { disable: true },
      control: { disable: true },
    },
    showRequiredMark: { table: { disable: false } },
  },
} satisfies Story;

export const WithHideLegend = {
  name: 'With HideLegend (FS-A7)',
  args: {
    ...defaultArgs,
    description: defaultDescription,
    helpText: defaultHelpText,
    hideLegend: true,
  },
  argTypes: {
    hideLegend: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const legend = canvas.getAllByText(defaultLegendText)[0];
    const descriptionNode = canvas.getByText(defaultDescription);
    const helpButtonNode = canvas.getByRole('button');
    await expect(legend).toBeInTheDocument();
    await expect(descriptionNode).toBeInTheDocument();
    await expect(helpButtonNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithSpacing = {
  name: 'With Spacing (A2)',
  args: {
    ...defaultArgs,
    hasSpacing: true,
  },
  argTypes: {
    hasSpacing: { table: { disable: false } },
  },
} satisfies Story;

export const WithDisabled = {
  name: 'With Disabled (A4)',
  args: {
    ...defaultArgs,
    disabled: true,
  },
  argTypes: {
    disabled: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const fieldset = canvas.getByRole('group');
    await expect(fieldset).toBeDisabled();
  },
} satisfies Story;

export const WithHelpTextSvgPathAndTitle = {
  name: 'With HelpText HelpSvgPath And TitleHelpSvg (FS-A1, FS-A5 delvis, FS-B2 delvis)',
  args: {
    ...defaultArgs,
    helpText: defaultHelpText,
    helpSvgPath: WarningSVGpath,
    titleHelpSvg: 'Tooltip',
  },
  argTypes: {
    helpText: { table: { disable: false } },
    helpSvgPath: { table: { disable: false } },
    titleHelpSvg: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      click: `${wrapper} > fieldset > button`,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button', {
      description: defaultLegendText,
    });
    await expect(helpButton).toBeInTheDocument();
    const svgNode = canvas.getByLabelText('Tooltip', { selector: 'svg' });
    await expect(svgNode).toBeInTheDocument();
    const legend = canvas.getAllByText(defaultLegendText)[0];
    await expect(helpButton).toHaveAttribute('aria-describedby', legend.id);
  },
} satisfies Story;

export const WithLongLegend = {
  name: 'With Long Legend',
  args: {
    ...defaultArgs,
    legend: loremIpsumWithoutSpaces,
  },
  argTypes: {
    legend: { table: { disable: false } },
  },
  parameters: {
    viewport: {
      defaultViewport: '--mobile',
    },
  },
} satisfies Story;

export const WithHelpToggleEvent = {
  name: 'With onHelpToggle Event',
  args: {
    ...defaultArgs,
    helpText: 'Hjelpetekst',
    onHelpToggle: (isOpen: boolean): void => {
      alert(isOpen ? 'Hjelpetekst blir vist' : 'Hjelpetekst skjules');
    },
  },
  parameters: {
    imageSnapshot: {
      disable: true,
    },
  },
} satisfies Story;

export const WithCustomClassNames = {
  name: 'With Custom ClassNames (FA3)',
  args: {
    ...defaultArgs,
    classNames: {
      legend: 'dummyClassname',
    },
  },
  argTypes: {
    classNames: {
      table: { disable: false },
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const legend = canvas.getAllByText(defaultLegendText)[1];
    await expect(legend).toHaveClass('dummyClassname');
  },
} satisfies Story;
