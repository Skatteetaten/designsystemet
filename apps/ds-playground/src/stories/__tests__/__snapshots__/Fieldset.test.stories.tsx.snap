// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Fieldset Defaults play-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:r2:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithAttributes play-test 1`] = `
<div data-test-block="true">
  <fieldset id="htmlid"
            data-testid="123ID"
            class="_fieldset_s6jvz_5  dummyClassname"
            lang="nb"
            form="123form"
  >
    <legend id="legendId-:r1:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithCustomClassNames play-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:rg:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15    dummyClassname"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithDescription play-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:r3:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15   _legendNoMarginBottom_s6jvz_28"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <div class="_description_10ldw_17">
      Kort hjelpetekst
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithDisabled play-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5"
            disabled
  >
    <legend id="legendId-:r9:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithHelpTextSvgPathAndTitle play-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:ra:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15   _legendNoMarginBottom_s6jvz_28"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <button class="_button_1hc5e_1 _button_extraSmall_1hc5e_34 _button_outlined_1hc5e_114 _helpButton_10ldw_27"
            data-brightness="default"
            type="button"
            aria-describedby="legendId-:ra:"
            aria-expanded="false"
    >
      <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hc5e_26"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rb:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rb:">
          Tooltip
        </title>
        <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
        </path>
      </svg>
    </button>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithHelpToggleEvent smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:re:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15   _legendNoMarginBottom_s6jvz_28"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <button class="_button_1hc5e_1 _button_extraSmall_1hc5e_34 _button_outlined_1hc5e_114 _helpButton_10ldw_27"
            data-brightness="default"
            type="button"
            aria-describedby="legendId-:re:"
            aria-expanded="false"
    >
      <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hc5e_26"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rf:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rf:">
          Hjelp
        </title>
        <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
        </path>
      </svg>
    </button>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithHideLegend play-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:r6:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15  _srOnly_s6jvz_39 _legendNoMarginBottom_s6jvz_28"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <button class="_button_1hc5e_1 _button_extraSmall_1hc5e_34 _button_outlined_1hc5e_114 _helpButton_10ldw_27 _srOnly_10ldw_59"
            data-brightness="default"
            type="button"
            aria-describedby="legendId-:r6:"
            aria-expanded="false"
    >
      <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_1hc5e_26"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r7:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r7:">
          Hjelp
        </title>
        <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
        </path>
      </svg>
    </button>
    <div class="_description_10ldw_17  _srOnly_10ldw_59">
      Kort hjelpetekst
    </div>
    <div class="_contentContainer_s6jvz_32 _contentContainerNoMarginTop_s6jvz_35">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithLongLegend smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:rd:"
            class="_srOnly_s6jvz_39"
    >
      Loremipsumdolorsitamet.Allesomharlagetennettside,trengtlittfylltekstellerbaresurfetrundtpånettetharantageligvissettdisseordene,etterfulgtaventilsynelatendeeviglangtekstfyltmedlatinskeliksomsetninger.
    </legend>
    <div class="_legend_s6jvz_15"
         aria-hidden="true"
    >
      Loremipsumdolorsitamet.Allesomharlagetennettside,trengtlittfylltekstellerbaresurfetrundtpånettetharantageligvissettdisseordene,etterfulgtaventilsynelatendeeviglangtekstfyltmedlatinskeliksomsetninger.
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithRef play-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5"
            id="dummyIdForwardedFromRef"
  >
    <legend id="legendId-:r0:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithShowRequiredMark smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:r4:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15 _legend_required_s6jvz_23"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithShowRequiredMarkAndLegend smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5">
    <legend id="legendId-:r5:"
            class="_srOnly_s6jvz_39"
    >
      <h1 class="_heading_12p5o_1 _heading_level3_12p5o_34">
        Dette er en Heading i legend
      </h1>
      <p class="_paragraph_1kzt5_1 _paragraph_ingress_1kzt5_20">
        <em>
          Dette er en italic Paragraph med ingress variant i legend
        </em>
      </p>
    </legend>
    <div class="_legend_s6jvz_15 _legendWithMarkup_required_s6jvz_23"
         aria-hidden="true"
    >
      <h1 class="_heading_12p5o_1 _heading_level3_12p5o_34">
        Dette er en Heading i legend
      </h1>
      <p class="_paragraph_1kzt5_1 _paragraph_ingress_1kzt5_20">
        <em>
          Dette er en italic Paragraph med ingress variant i legend
        </em>
      </p>
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithSpacing smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="_fieldset_s6jvz_5 _fieldset_hasSpacing_s6jvz_11">
    <legend id="legendId-:r8:"
            class="_srOnly_s6jvz_39"
    >
      Ledetekst
    </legend>
    <div class="_legend_s6jvz_15"
         aria-hidden="true"
    >
      Ledetekst
    </div>
    <div class="_contentContainer_s6jvz_32">
      <div>
        Innhold
      </div>
    </div>
  </fieldset>
</div>
`;
